fix(orchestrator): restore Agent typecheck compatibility for Smithers response recovery

## Problem

The current `develop` branch fails the Agent package typecheck in
`plugins/plugin-agent-orchestrator/src/services/smithers-task-integration.ts`:

```text
smithers-task-integration.ts(258,6): error TS2550: Property 'findLast' does not exist on type 'unknown[]'.
smithers-task-integration.ts(259,8): error TS7006: Parameter 'value' implicitly has an 'any' type.
```

The orchestrator package itself declares an `ESNext` lib, but its source is also
compiled through the Agent workspace boundary. That consuming TypeScript
configuration does not expose `Array.prototype.findLast`, so one ES2023 array
method breaks `bun run --cwd packages/agent typecheck` even though the runtime
logic only needs a small reverse search.

Changing the shared consuming lib target would broaden the runtime contract for
an otherwise local response-recovery operation. The narrow repair is to scan the
durable turn outputs from newest to oldest without relying on `findLast`.

## Scope

- `plugins/plugin-agent-orchestrator/src/services/smithers-task-integration.ts`
- focused Smithers task-integration regression coverage

No Smithers persistence schema, task lifecycle, ACP transport, Agent tsconfig,
or public API change is in scope.

## Acceptance criteria

- Durable response recovery selects the newest non-empty string output exactly
  as the current `findLast` predicate intends.
- Recovery remains correct across a completed multi-turn run and a subsequent
  restart/resume without another ACP prompt.
- The implementation uses no ES2023-only `Array.prototype.findLast` dependency
  at the Agent source-consumption boundary.
- Focused orchestrator tests, orchestrator and Agent typecheck, lint/build,
  `git diff --check`, and root verification run on the exact rebased head;
  unrelated failures are recorded precisely.

## Evidence plan

- UI screenshots/video/OCR: N/A - TypeScript/runtime service only; no rendered surface.
- Frontend logs: N/A - no frontend code or state.
- Real-LLM trajectory: N/A - deterministic durable-run recovery; no live model is required.
- Backend/domain artifact: focused transcript proving newest-response recovery
  before restart and after restart with zero replayed ACP prompts, plus exact
  Agent/orchestrator typecheck output.

## Contribution provenance

- AI assistance: yes
- AI provider/model: openai / gpt-5.6-sol
- Client / agent tooling: Codex desktop
- Contribution skill revision: elizaOS/eliza@78bd11ff6c23234edf5c6ed712ef694661a1645a:packages/skills/skills/contribute-to-eliza
- Attribution status: self-reported

— [codex-smithers-response-compat]
<!-- eliza-computer-attribution:v1 {"provider":"openai","model":"gpt-5.6-sol","client":"Codex desktop","skill_revision":"elizaOS/eliza@78bd11ff6c23234edf5c6ed712ef694661a1645a:packages/skills/skills/contribute-to-eliza"} -->

--- a/plugins/plugin-agent-orchestrator/__tests__/unit/smithers-task-integration.test.ts
+++ b/plugins/plugin-agent-orchestrator/__tests__/unit/smithers-task-integration.test.ts
@@ -73,14 +73,19 @@ describe("runDurableTask", () => {
   );
 
   it(
-    "recovers a completed response without another ACP prompt after restart",
+    "recovers the newest completed response without another ACP prompt after restart",
     async () => {
       const session = uniqueSession();
       const durableResponse = `durable-${"x".repeat(128 * 1024)}-tail`;
-      const sendPrompt = vi.fn(async () => ({
-        stopReason: "end_turn",
-        finalText: durableResponse,
-      }));
+      const responses = [
+        { stopReason: "max_tokens", finalText: "older partial response" },
+        { stopReason: "end_turn", finalText: durableResponse },
+      ];
+      const sendPrompt = vi.fn(async () => {
+        const response = responses.shift();
+        if (!response) throw new Error("unexpected extra ACP prompt");
+        return response;
+      });
       const service: AcpTaskService = {
         sendPrompt,
         cancelSession: cancellation(),
@@ -88,17 +93,21 @@ describe("runDurableTask", () => {
 
       const first = await runDurableTask(service, session, "do it once", {
         tenantId,
+        maxTurns: 2,
       });
       expect(first.lastResponse).toBe(durableResponse);
-      expect(sendPrompt).toHaveBeenCalledTimes(1);
+      expect(first.turns).toBe(2);
+      expect(sendPrompt).toHaveBeenCalledTimes(2);
 
       sendPrompt.mockClear();
       const resumed = await runDurableTask(service, session, "do it once", {
         tenantId,
+        maxTurns: 2,
       });
       expect(sendPrompt).not.toHaveBeenCalled();
       expect(resumed.lastResponse).toBe(durableResponse);
       expect(resumed.status).toBe("completed");
+      expect(resumed.turns).toBe(2);
     },
     TIMEOUT,
   );

--- a/plugins/plugin-agent-orchestrator/src/services/smithers-task-integration.ts
+++ b/plugins/plugin-agent-orchestrator/src/services/smithers-task-integration.ts
@@ -56,6 +56,16 @@ function nonEmptyString(value: unknown): value is string {
   return typeof value === "string" && value.trim().length > 0;
 }
 
+function findLastNonEmptyString(
+  values: readonly unknown[],
+): string | undefined {
+  for (let index = values.length - 1; index >= 0; index -= 1) {
+    const value = values[index];
+    if (nonEmptyString(value)) return value;
+  }
+  return undefined;
+}
+
 function approvalPreset(value: unknown): ApprovalPreset | undefined {
   return value === "readonly" ||
     value === "standard" ||
@@ -253,12 +263,11 @@ export async function runDurableTask(
       severity: "ephemeral",
     });
   }
-  const recoveredResponse = collectDurableTaskTurns(result.execution)
-    .map((turn) => turn.output?.finalText)
-    .findLast(
-      (value): value is string =>
-        typeof value === "string" && value.trim().length > 0,
-    );
+  const recoveredResponse = findLastNonEmptyString(
+    collectDurableTaskTurns(result.execution).map(
+      (turn) => turn.output?.finalText,
+    ),
+  );
   const lastResponse = executor.lastResponse ?? recoveredResponse;
   if (typeof lastResponse !== "string" || lastResponse.trim().length === 0) {
     throw new ElizaError("Durable task completed without a response", {

